tests/_client.lua

```diff
@@ -0,0 +1,33 @@
+local spawn = require("awful.spawn")
+
+-- This file provide a simple, yet flexible, test client.
+-- It is used to test the `awful.rules`
+
+return function(class, title)
+    title = title or 'Awesome test client'
+
+    local cmd = table.concat {
+        "lua -e \"",
+        "local lgi = require 'lgi';",
```

Comment:
Could you make this:
`local cmd = { "lua", "-e", table.concat { "here is the actual code" }}`? That way we don't have useless argument splitting going on and the code doesn't need to be enclosed in `"`.

You could also make this
```
local code = [[
local lgi = require('lgi');
etc
print('class is', 
]] .. class .. [[ , 'and Uli agrees that this part is ugly ]]
```